packages/arb-bridge-eth/contracts/rollup/Node.sol

```diff
@@ -17,131 +17,146 @@
  */
 
 pragma solidity ^0.6.11;
+pragma experimental ABIEncoderV2;
 
-import "./INode.sol";
-import "../libraries/Cloneable.sol";
 import "@openzeppelin/contracts/math/SafeMath.sol";
 
-contract Node is Cloneable, INode {
-    using SafeMath for uint256;
+struct NodeProps {
+    // Hash of the state of the chain as of this node
+    bytes32 stateHash;
 
-    /// @notice Hash of the state of the chain as of this node
-    bytes32 public override stateHash;
+    // Hash of the data that can be challenged
+    bytes32 challengeHash;
 
-    /// @notice Hash of the data that can be challenged
-    bytes32 public override challengeHash;
+    // Hash of the data that will be committed if this node is confirmed
+    bytes32 confirmData;
 
-    /// @notice Hash of the data that will be committed if this node is confirmed
-    bytes32 public override confirmData;
+    // Index of the node previous to this one
+    uint256 prev;
```

Comment:
we could instead define `prev` as a library function that actually gets the previous node struct.
we could instead rename this to prevNum or something

Reply:
a bit gimmicky, but could be neat

Reply:
We actually do quite a bit of node number comparing with it, so I think it's good to keep it there. Will rename to prevNum